mypy/test/teststubgen.py

```diff
@@ -218,6 +218,19 @@ def test_infer_sig_from_docstring(self) -> None:
 
         assert_equal(infer_sig_from_docstring('\nfunc[(x: foo.bar, invalid]', 'func'), [])
 
+        assert_equal(infer_sig_from_docstring('\nfunc(x: invalid::type<with_template>)', 'func'),
+                     [FunctionSig(name='func', args=[ArgSig(name='x', type=None)],
+                                  ret_type='Any')])
+
+        assert_equal(infer_sig_from_docstring('\nfunc(x: str="")', 'func'),
+                     [FunctionSig(name='func', args=[ArgSig(name='x', type='str', default=True)],
+                                  ret_type='Any')])
+
+    def test_infer_sig_from_docstring_duplicate_args(self) -> None:
+        assert_equal(infer_sig_from_docstring('\nfunc(x, x) -> str\nfunc(x, y) -> int', 'func'),
```

Comment:
I take it there's some kind of bug in pybind11 that sometimes generates `func(x, x)`?

Reply:
There is a bug in pybind11 that sometimes generates signatures using C++ types instead of Python types. As I tried some sanitizations I got errors for duplicate arguments. Though right now those invalid signatures are just dropped, so this is not longer the case